src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/AbstractCodeMetricsWriterService.java

```diff
@@ -12,55 +15,67 @@
+    protected boolean doesItMatterToDisplay(CodeMetricsDiffResult codeMetricsDiffResult, List<String> methodMetricsToDisplay) {
+        Boolean doesItMatter = codeMetricsDiffResult.getOldValue() != codeMetricsDiffResult.getNewValue();
+        return shouldWeDisplay(methodMetricsToDisplay, codeMetricsDiffResult) && doesItMatter;
     }
 
-    protected boolean shouldWeDisplay(List<String> listOfMetricsToDisplay, CodeMetricsResult codeMetricsResult) {
-        return listOfMetricsToDisplay.contains(codeMetricsResult.getMetricName()) || listOfMetricsToDisplay.isEmpty();
+    protected boolean shouldWeDisplay(List<String> listOfMetricsToDisplay, CodeMetricsDiffResult codeMetricsDiffResult) {
+        return listOfMetricsToDisplay.contains(codeMetricsDiffResult.getMetricName()) || listOfMetricsToDisplay.isEmpty();
     }
 
-    protected List<String> getMetricsToDisplayFromConfig(List<String> methodConfig) {
-        return methodConfig.stream().filter(metricToDisplay -> !metricToDisplay.startsWith("#")).collect(Collectors.toList());
+    protected List<String> getMetricsToDisplayFromConfig(List<String> config) {
+        return config.stream().filter(metricToDisplay -> !metricToDisplay.startsWith("#")).collect(Collectors.toList());
     }
 
-    protected CSVPrinter getCsvPrinterVertical(char delimiter, StringWriter sw) throws IOException {
-        CSVFormat csvFormat = CSVFormat.DEFAULT.withHeader(
-                "FILE", "CLASS", "TYPE", "METRIC", "NEW_VALUE", "OLD_VALUE"
-        ).withDelimiter(delimiter);
-
-        return new CSVPrinter(sw, csvFormat);
+    protected CSVPrinter getVerticalPrinterWithDelimiter(String format, StringWriter sw) throws IOException {
+        try {
+            char delimiter =  ProgramConstants.DelimeterForReport.valueOf(format.toUpperCase()).getDelimeter();
+            CSVFormat csvFormat = CSVFormat.DEFAULT.withHeader(
+                    "FILE", "CLASS", "TYPE", "METRIC", "NEW_VALUE", "OLD_VALUE"
+            ).withDelimiter(delimiter);
+            return new CSVPrinter(sw, csvFormat);
+        } catch(IllegalArgumentException e) {
+            return null;
+        }
     }
 
-    protected CSVPrinter getCsvPrinterHorizontal(char delimiter, StringWriter sw) throws IOException {
+    protected CSVPrinter getHorizontalPrinterWithDelimiter(String format, StringWriter sw) throws IOException {
+        char delimiter =  ProgramConstants.DelimeterForReport.valueOf(format.toUpperCase()).getDelimeter();
         CSVFormat csvFormat = CSVFormat.DEFAULT.withDelimiter(delimiter);
         return new CSVPrinter(sw, csvFormat);
     }
 
+    protected Builder getVerticalMarkdownPrinter() {
+        Builder tableBuilder = new Builder();
+        tableBuilder.withAlignments(Table.ALIGN_CENTER, Table.ALIGN_CENTER, Table.ALIGN_CENTER, Table.ALIGN_CENTER, Table.ALIGN_CENTER, Table.ALIGN_CENTER);
+        tableBuilder.addRow("FILE", "CLASS", "TYPE", "METRIC", "NEW_VALUE", "OLD_VALUE");
```

Comment:
If field list is same for getVerticalPRinterWithDelimiter and here, consider extracting the list.

Reply:
Thanks for identifying this, I have now abstracted it out and used that method, please check the latest commit